plink_clump_hail.py

```diff
@@ -0,0 +1,366 @@
+                   alleles = hl.array([ht.varid.split(':')[2],
+                                       ht.varid.split(':')[3]])
+    )
+    ht = ht.annotate_globals(trait_type = args.trait_type,
+                             phenocode = args.phenocode,
+                             pheno_sex = args.pheno_sex,
+                             coding = args.coding,
+                             modifier = args.modifier)
+    ht = ht.drop('contig','varid','pos')
+    ht.describe()
+#    ht.select().show()
+    ht.write(args.output_file, overwrite=args.overwrite)
+
+        
+def export_ma_format(batch_size=256):
+    r'''
+    Export columns for .ma format (A1, A2, freq, beta, se, N) for select phenotypes
+    '''
+    meta_mt0 = hl.read_matrix_table('gs://ukb-diverse-pops/combined_results/meta_analysis.mt')
+    
+    highprev = hl.import_table(f'{ldprune_dir}/joined_ukbb_lancet_age_high_prev.tsv', impute=True)
+    highprev = highprev.annotate(pheno = highprev.code.replace('_irnt',''))
+    pheno_list = highprev.pheno.collect()
+    pheno_list = [p for p in pheno_list if p is not None]
+    meta_mt0 = meta_mt0.filter_cols(hl.literal(pheno_list).contains(meta_mt0.pheno))
+
+    meta_mt0 = meta_mt0.annotate_cols(pheno_id = (meta_mt0.trait_type+'-'+
+                                      meta_mt0.phenocode+'-'+
+                                      meta_mt0.pheno_sex+
+                                      hl.if_else(hl.len(meta_mt0.coding)>0, '-'+meta_mt0.coding, '')+
+                                      hl.if_else(hl.len(meta_mt0.modifier)>0, '-'+meta_mt0.modifier, '')
+                                      ).replace(' ','_').replace('/','_'))
+    
+    meta_mt0 = meta_mt0.annotate_rows(SNP = meta_mt0.locus.contig+':'+hl.str(meta_mt0.locus.position)+':'+meta_mt0.alleles[0]+':'+meta_mt0.alleles[1],
+                                      A1 = meta_mt0.alleles[1], # .ma format requires A1 = effect allele, which in this case is A2 for UKB GWAS
+                                      A2 = meta_mt0.alleles[0])
+
+    meta_field_rename_dict = {'BETA':'b',
+                          'SE':'se',
+                          'Pvalue':'p',
+                          'AF_Allele2':'freq',
+                          'N':'N'}
+    
+    all_pops = ['AFR', 'AMR', 'CSA', 'EAS', 'EUR', 'MID']
+
+    for pop in ['AFR','EUR']: #['AFR','AMR','CSA','EAS','EUR','MID']:
+        print(f'not_{pop}')
+
+        req_pop_list = [p for p in all_pops if p is not pop]
+
+        loo_pop = meta_mt0.annotate_cols(idx = meta_mt0.meta_analysis_data.pop.index(hl.literal(req_pop_list))) # get index of which meta-analysis is the leave-on-out for current pop
+        loo_pop = loo_pop.filter_cols(hl.is_defined(loo_pop.idx))
+        
+        annotate_dict = {}
+        for field in ['AF_Allele2','BETA','SE','Pvalue','N']:
+            annotate_dict.update({meta_field_rename_dict[field]: loo_pop.meta_analysis[field][loo_pop.idx]}) 
```

Comment:
`annotate_dict = {meta_field_rename_dict[field]: loo_pop.meta_analysis[field][loo_pop.idx] for field in ['AF...']`